### PR TITLE
Mongoose

### PR DESCRIPTION
**Pull request by jeversmann**
Added Mongoose Schemas in `/models` and changed everything else to use the Mongoose models in order to hit mongo.

Addresses #3, #4, and #11.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -4,10 +4,28 @@ var favicon = require('serve-favicon');
...
+var config = require('./config.json');
 
-var mongo = require('mongodb')
-  , monk  = require('monk');
-var db = monk('localhost:27017/utcs-uptime');
+// Connect database
+var mongoose = require('mongoose');
+var connect = function () {
+    var options = { server: { socketOptions: { keepAlive: 1 } } };
+    mongoose.connect(config.database, options);
```
Comment (JDong820) [inline app.js]: wow node is so easy, `config.*`

File: app.js [hunk 2]
```diff
@@ -4,10 +4,28 @@ var favicon = require('serve-favicon');
...
+};
+connect();
+
+mongoose.connection.on('error', console.error.bind(console, 'connection error:'));
```
Comment (JDong820) [inline app.js]: does this print "connection error:" on errors?

File: background.js [hunk 3]
```diff
@@ -31,14 +17,18 @@ fs.readFile(args[1], 'utf-8', function (err, data) {
     if (!server) {
```
Comment (JDong820) [inline background.js]: I don't understand what `server` represents or where it is defined

File: models/session.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+var mongoose = require('mongoose');
+
+var sessionSchema = mongoose.Schema({
+  name: {type: String, default: '', trim: true},
+  hostname: {type: String, default: '', trim: true},
+  from: {type: String, default: '', trim: true},
+  loginTime: {type: Date},
+  logoutTime: {type: Date, default: Date.now},
```
Comment (JDong820) [inline models/session.js]: So we update entries when we realize they haven't logged out. Sure.

File: models/session.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+var mongoose = require('mongoose');
+
+var sessionSchema = mongoose.Schema({
+  name: {type: String, default: '', trim: true},
```
Comment (JDong820) [inline models/session.js]: Why are there default names

File: models/current.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+var mongoose = require('mongoose');
+
+var currentSchema = mongoose.Schema({
+  hostname: {type: String, default: '', trim: true},
```
Comment (JDong820) [inline models/current.js]: why are there default hostnames? is this for debugging?

File: package.json [hunk 7]
```diff
@@ -18,15 +18,15 @@
...
-    "morgan": "~1.3.0",
-    "serve-favicon": "~2.1.3",
     "debug": "~2.0.0",
+    "express": "~4.9.0",
     "jade": "~1.6.0",
-    "monk": "*",
-    "mongodb": "~2.0"
+    "mongodb": "~2.0",
+    "mongoose": "^3.8.21",
+    "morgan": "~1.3.0",
```
Comment (JDong820) [inline package.json]: Why do we still need morgan? For logging?

File: routes/current.js [hunk 8]
```diff
@@ -18,8 +21,8 @@ router.route('/')
 // This should only ever return one
```
Comment (JDong820) [inline routes/current.js]: Should we enforce that? With an err?

File: who.js [hunk 9]
```diff
@@ -3,19 +3,27 @@ var exec = require('child_process').exec;
...
+
+  var users = lines.slice(2, -1).map(
     function (line) {
-      var who = {};
-      line.split(/\s+/).forEach(
-        function(value, index) {
-          who[names[index]] = value;
-      });
-      return who;
+    var who = {};
```
Comment (JDong820) [inline who.js]: Is this the meta for indent style

File: who.js [hunk 10]
```diff
@@ -24,10 +32,12 @@ function ssh (username, server, callback) {
     throw new Error('Invalid ssh operands');
   }
   // Build and exec a command if we have the stuff
-  exec('ssh ' + username + '@' + server + ' w',
-       function (error, stdout, stderr) {
-         if (error) {
-           throw error;
+  exec('ssh ' + username + '@' + server + ' w -s',
```
Comment (JDong820) [inline who.js]: <pre>
-s, --short
              Use the short format.  Don't print the login time, JCPU or PCPU times.
</pre>
I thought we wanted the login time? I guess we could infer it from `IDLE`

**Comment by JDong820** [comment]
Ok, looks like
* added config file
* switched to mongoose
No tests broke, and I'm asserting you've tested this to work. Some questions about the code, but it should be good.

File: models/session.js (hunk 4 above)
Comment (jeversmann) [inline models/session.js]: I don't understand.

File: package.json (hunk 7 above)
Comment (jeversmann) [inline package.json]: There is a dev dependency section, and I can move morgan if you want.

File: routes/current.js (hunk 8 above)
Comment (jeversmann) [inline routes/current.js]: It would be more logical to switch `find` to `findOne` because Mongoose can do that for us.

File: who.js (hunk 10 above)
Comment (jeversmann) [inline who.js]: We can't actually infer it, so I'll put it back.

File: app.js (hunk 1 above)
Comment (jeversmann) [inline app.js]: I'm just hoping it can be stubbed in tests the way I want.

File: background.js (hunk 3 above)
Comment (jeversmann) [inline background.js]: Are you looking at the whole diff? `server` is defined literally one line up.

**Comment by jeversmann** [comment]
The default values in the Schemas don't really have a reason for them. Should I make them all more obvious values so we'll notice if they ever get introduced?

File: package.json (hunk 7 above)
Comment (JDong820) [inline package.json]: We should, since morgan isn't used for anything else and I don't foresee it being used for anything else.

**Comment by JDong820** [comment]
If you don't have a reason for default values, you shouldn't have them.

File: models/session.js (hunk 4 above)
Comment (JDong820) [inline models/session.js]: It concerned me that you had a non-fixed value as a default, but this one looks okay.

**Comment by jeversmann** [comment]
Ok, I'll trim the defaults and go straight for validations.

**Pull request merged**